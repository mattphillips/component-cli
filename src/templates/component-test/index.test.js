import each from 'jest-each';
import componentTest from './';

describe('Component Test', () => {
  it('renders with correct component name', () => {
    expect(componentTest({ name: 'Header' })).toMatchSnapshot();
  });

  each([
    ['index', { name: 'Header', namingConvention: 'index.js|x' }],
    ['component name', { name: 'Header', namingConvention: 'ComponentName.js|x' }]
  ]).it('build template with correct import: %s', (description, options) => {
    expect(componentTest(options)).toMatchSnapshot();
  });
});
