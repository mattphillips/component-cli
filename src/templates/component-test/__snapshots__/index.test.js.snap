// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Test build template with correct import: component name 1`] = `
"import React from 'react';
import { shallow } from 'enzyme';

import Header from './Header';

describe('<Header />', () => {
  it('renders with props', () => {
    const props = {};
    const component = shallow(<Header {...props} />);
    expect(component).toMatchSnapshot();
  });
});
"
`;

exports[`Component Test build template with correct import: index 1`] = `
"import React from 'react';
import { shallow } from 'enzyme';

import Header from './';

describe('<Header />', () => {
  it('renders with props', () => {
    const props = {};
    const component = shallow(<Header {...props} />);
    expect(component).toMatchSnapshot();
  });
});
"
`;

exports[`Component Test renders with correct component name 1`] = `
"import React from 'react';
import { shallow } from 'enzyme';

undefined;

describe('<Header />', () => {
  it('renders with props', () => {
    const props = {};
    const component = shallow(<Header {...props} />);
    expect(component).toMatchSnapshot();
  });
});
"
`;
